tools/test-backend

```diff
@@ -168,10 +196,24 @@ if __name__ == "__main__":
 
     TestRunner = get_runner(settings)
     test_runner = TestRunner()
-    failures = test_runner.run_tests(suites, fatal_errors=options.fatal_errors,
-                                     full_suite=full_suite)
+    failures, failed_tests = test_runner.run_tests(suites, fatal_errors=options.fatal_errors,
+                                                   full_suite=full_suite)
 
     templates_not_rendered = test_runner.get_shallow_tested_templates()
+
+    # If one or more tests fail, we store the test as a JSON list
+    # at zulip/var/last_test_failure.json so that the failed test
+    # can be run alone next time using the command --re-run
+    if failures:
+        failed_test_name = {}
+        failed_test_name['name'] = failed_tests
+        try:
+            with open(failed_test_path, 'w') as outfile:
+                ujson.dump(failed_test_name, outfile)
+        except EnvironmentError as exception:
+            print('Oops, we got %s while tring to write the data at %s \
+            Plese rectify it and try again.' % (exception, failed_test_path))
```

Comment:
I'd expect an IOError or OSError in that case.  It's probably fine to not catch an exception for now and add code of this form if something shows up.